coinparam/register.go

```diff
@@ -31,6 +32,26 @@ type Params struct {
 
 	// GenesisHash is the starting block hash.
 	GenesisHash *chainhash.Hash
+	
+	// The function used to calculate the proof of work value for a block
+	PoWFunction func(b []byte) chainhash.Hash
+  
+    // The function used to calculate the difficulty of a given block
+    DiffCalcFunction func(r io.ReadSeeker, height, startheight int32, p *Params) (uint32, error)
```

Comment:
Height here is just to inform the difficulty function which block it needs to calculate the difficulty of. The other alternative is to pass the header directly into the function, but this would not work for LTC or VTC which have a variable retarget lookback.